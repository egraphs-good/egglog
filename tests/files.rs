use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
