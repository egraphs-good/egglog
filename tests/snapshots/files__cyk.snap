---
source: tests/files.rs
expression: serialized
---
nodes:
  Unit-0:
    op: ()
    children: []
    eclass: Unit-0
    cost: 0
  Prod-7659469028595837896:
    op: Prod
    children:
      - NonTerm-9670903108499433938
      - NonTerm-7415288725214994367
      - NonTerm-1566882640312150612
    eclass: Unit-0
    cost: 1
  Prod-1081172882011038115:
    op: Prod
    children:
      - NonTerm-9670903108499433938
      - NonTerm-1566882640312150612
      - NonTerm-14842264911260568083
    eclass: Unit-0
    cost: 1
  Prod-6544191068063263740:
    op: Prod
    children:
      - NonTerm-7415288725214994367
      - NonTerm-1566882640312150612
      - NonTerm-7415288725214994367
    eclass: Unit-0
    cost: 1
  Prod-14534116445915336970:
    op: Prod
    children:
      - NonTerm-1566882640312150612
      - NonTerm-14842264911260568083
      - NonTerm-14842264911260568083
    eclass: Unit-0
    cost: 1
  Prod-15206526507414598257:
    op: Prod
    children:
      - NonTerm-14842264911260568083
      - NonTerm-7415288725214994367
      - NonTerm-1566882640312150612
    eclass: Unit-0
    cost: 1
  String-17627030480522586708:
    op: "\"a\""
    children: []
    eclass: String-17627030480522586708
    cost: 0
  End-11856760466294172486:
    op: End
    children:
      - NonTerm-7415288725214994367
      - String-17627030480522586708
    eclass: Unit-0
    cost: 1
  String-3799122101935431485:
    op: "\"b\""
    children: []
    eclass: String-3799122101935431485
    cost: 0
  End-5711696037964013857:
    op: End
    children:
      - NonTerm-1566882640312150612
      - String-3799122101935431485
    eclass: Unit-0
    cost: 1
  End-14321860509961492610:
    op: End
    children:
      - NonTerm-14842264911260568083
      - String-17627030480522586708
    eclass: Unit-0
    cost: 1
  String-9670903108499433938:
    op: "\"S\""
    children: []
    eclass: String-9670903108499433938
    cost: 0
  NonTerm-9670903108499433938:
    op: NonTerm
    children:
      - String-9670903108499433938
    eclass: "0"
    cost: 1
  String-7415288725214994367:
    op: "\"A\""
    children: []
    eclass: String-7415288725214994367
    cost: 0
  NonTerm-7415288725214994367:
    op: NonTerm
    children:
      - String-7415288725214994367
    eclass: "1"
    cost: 1
  String-1566882640312150612:
    op: "\"B\""
    children: []
    eclass: String-1566882640312150612
    cost: 0
  NonTerm-1566882640312150612:
    op: NonTerm
    children:
      - String-1566882640312150612
    eclass: "2"
    cost: 1
  String-14842264911260568083:
    op: "\"C\""
    children: []
    eclass: String-14842264911260568083
    cost: 0
  NonTerm-14842264911260568083:
    op: NonTerm
    children:
      - String-14842264911260568083
    eclass: "3"
    cost: 1
root_eclasses: []
class_data:
  Unit-0:
    type: Unit
  "0":
    type: nonterm
  "1":
    type: nonterm
  "2":
    type: nonterm
  "3":
    type: nonterm
  String-17627030480522586708:
    type: String
  String-3799122101935431485:
    type: String
  String-9670903108499433938:
    type: String
  String-7415288725214994367:
    type: String
  String-1566882640312150612:
    type: String
  String-14842264911260568083:
    type: String

